cylc/flow/scripts/broadcast.py

```diff
@@ -421,20 +421,24 @@ async def run(options: 'Values', workflow_id):
         report_set = True
 
     results = await pclient.async_request('graphql', mutation_kwargs)
-    for result in results['broadcast']['result']:
-        modified_settings = result['response'][0]
-        bad_options = result['response'][1]
-        if modified_settings:
-            ret['stdout'].append(
-                get_broadcast_change_report(
-                    modified_settings,
-                    is_cancel=report_cancel,
+    try:
+        for result in results['broadcast']['result']:
+            modified_settings = result['response'][0]
+            bad_options = result['response'][1]
+            if modified_settings:
+                ret['stdout'].append(
+                    get_broadcast_change_report(
+                        modified_settings,
+                        is_cancel=report_cancel,
+                    )
                 )
-            )
+        bad_result = report_bad_options(bad_options, is_set=report_set)
+    except TypeError:
```

Comment:
Is it necessary to catch `TypeError`s for all these lines? Could it catch more than anticipated?

Reply:
Well, I believe it's the `for` line that's the issue.. When results isn't a dictionary.. so I could test for that..

Reply:
Confirmed, when results is an error list and not a dictionary it fails (other errors still caught normally):
![image](https://user-images.githubusercontent.com/11400777/192660671-2afba9a1-6059-47f5-ab2e-00565cd5419e.png)
This is for internal errors that would otherwise be lost.

I could test that results is a dictionary, instead, but if somehow it catches more than anticipated at least the end user would have something to report back to us (instead of just a `TypeError` error in the script)..